core/lib/page-functions.js

```diff
@@ -539,35 +539,25 @@ function getNodeDetails(element) {
   return details;
 }
 
-const getNodeDetailsString = `function getNodeDetails(element) {
-  ${getNodePath.toString()};
-  ${getNodeSelector.toString()};
-  ${getBoundingClientRect.toString()};
-  ${getOuterHTMLSnippet.toString()};
-  ${getNodeLabel.toString()};
-  return (${getNodeDetails.toString()})(element);
-}`;
-
-// TODO(esmodules): should this be refactored to export each function individually?
+getNodeDetails.dependencies = [
+  getNodePath,
+  getNodeSelector,
+  getBoundingClientRect,
+  getOuterHTMLSnippet,
+  getNodeLabel,
+];
```

Comment:
What if we just override the `toString` function on `getNodeDetails` to include the deps? It's the only place we would need to do it.

Reply:
Tacking on a property to Function is iffy enough, I draw the line at overriding `Object.prototype.toString` :)

Reply:
In the realm of js hackery, though, it does seem much more idiomatic to override a function's `toString` to customize what happens when it's serialized to a string than to have a magic property for it. That's `toString`'s literal purpose.

Reply:
> Tacking on a property to Function is iffy enough, I draw the line at overriding Object.prototype.toString

I wasn't suggesting overriding `Object.prototype.toString`. I was suggesting we do `getNodeDetails.toString = () => {...}` or something

Reply:
I know, s/override/shadow